src/posit/connect/bundles.py

```diff
@@ -99,14 +96,66 @@ def delete(self) -> None:
         url = urls.append(self.config.url, path)
         self.session.delete(url)
 
+    def download(self) -> bytes:
+        """Download a bundle.
+
+        Returns
+        -------
+        bytes
+            Archive contents in bytes representation.
+
+        Examples
+        --------
+        >>> with open('archive.tar.gz', 'wb') as file:
+        >>>     data = bundle.download()
+        >>>     file.write(data)
+        """
+        path = f"v1/content/{self.content_guid}/bundles/{self.id}/download"
+        url = urls.append(self.config.url, path)
+        response = self.session.get(url, stream=True)
+        return response.content
+
 
-class Bundles(Resources):
+class Bundles(resources.Resources):
     def __init__(
-        self, config: Config, session: Session, content_guid: str
+        self, config: config.Config, session: Session, content_guid: str
     ) -> None:
         super().__init__(config, session)
         self.content_guid = content_guid
 
+    def create(self, data: BinaryIO | bytes) -> Bundle:
+        """Create a bundle.
+
+        Create a bundle by upload via archive format.
+
+        Parameters
+        ----------
+        data : BinaryIO | bytes
+            Archive contents in BinaryIO or bytes representation.
```

Comment:
Similar to above, would it make sense be able to just pass a file path? I would imagine that `requests` would have some handling for that kind of form posting too.

Reply:
Yes, looks like requests supports providing a filename as a string. I'll try that out.

Reply:
There is precedent in a number of libraries for functions that accept a file handle or filename.

Reply:
Do you know any off the top of your head?

Reply:
https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html is an example